tiledb/sm/filesystem/vfs.h

```diff
@@ -84,14 +84,46 @@ enum class VFSMode : uint8_t;
 
 /** The VFS configuration parameters. */
 struct VFSParameters {
+  /* ********************************* */
+  /*          TYPE DEFINITIONS         */
+  /* ********************************* */
+
+  enum struct ReadLoggingMode {
+    DISABLED,
+    FRAGMENTS,
+    FRAGMENT_FILES,
```

Comment:
I'm not sure what you mean here. I can kind of see what you mean by the VFS knowing what a fragment is, but given that this is just logging for analysis I'm not too bothered by it. With the changes suggested by @eric-hughes-tiledb it'll not even be in the VFS class any more.

For the two config variables, I'm not entirely sure how they would replace the existing behavior of what granularity to log and/or control the logic for only logging on first sight of the URI. We also don't need writes or deletes since both of those are trivially observable from the filesystem/object store. Perhaps if we had any sort of update-in-place semantics writes would be useful but we're write-once-never-modify across the board.

I'm not 100% sure how you mean to use `vfs.log_files`. If you mean "only log URIs matching this pattern", I'm not coming up with anything useful given how random and spartan our object names are. The current "is_fragment_related" check could maybe be extended to other major directories but given those are relatively small in comparison to fragments I don't see the benefit in proactively adding them. If someone does end up needing them we can also add more modes to select them. On the other hand if you mean specifying a pattern for what to log akin to specifying a format string for something like [std::format](https://en.cppreference.com/w/cpp/utility/format/format) that's a whole can of worms I'd rather not open.

Reply:
Makes sense, thanks!